srml/babe/src/lib.rs

```diff
@@ -151,16 +170,38 @@ decl_module! {
 				.iter()
 				.filter_map(|s| s.as_pre_runtime())
 				.filter_map(|(id, mut data)| if id == BABE_ENGINE_ID {
-					<[u8; VRF_OUTPUT_LENGTH]>::decode(&mut data)
+					<(u64, [u8; VRF_OUTPUT_LENGTH])>::decode(&mut data)
 				} else {
 					None
 				}) {
-				Self::deposit_vrf_output(&i);
+				let slot_index = i.0
+					.checked_add(1)
+					.expect("slot numbers will never reach 2^64 in our lifetimes; qed");
+				let epoch_change_slot = slot_index
+					.checked_add(Self::slots_per_epoch())
```

Comment:
Isn't slot_index the current slot? How does this work? Shouldn't this be `slot_index % slots_per_epoch == 0` instead?

Reply:
`slot_index` is indeed the current slot, and so this makes `epoch_change_slot` the slot that the next epoch change will happen at.  That said, your version is cleaner.

Reply:
I think @andresilva is right and your current code just does not work, how should both values be ever equal, when the second is the first + some value?

Reply:
Fixed (in a different way).